sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -71,7 +71,9 @@ class DataFrameReader private[sql](sqlContext: SQLContext) extends Logging {
    * @since 1.4.0
    */
   def option(key: String, value: String): DataFrameReader = {
-    this.extraOptions += (key -> value)
+    if (value != null) {
```

Comment:
The point is to be able to set the option to null. With spark-csv, if some options are null instead of not set, it changes the behavior. This denatures the pull request I made.